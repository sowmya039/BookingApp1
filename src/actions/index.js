import constants from '../constants';

function getSavedRooms(rooms = []) {
  return {
    type: constants.GET_SAVED_ROOMS,
  };
}

function updateRooms(ev, index) {
  return {
    type: constants.UPDATE_ROOMS,
    payload: {
      ev,
      index
    }
  };
}

function saveRooms() {
  return {
    type: constants.SAVE_ROOMS
  };
}
export { getSavedRooms, updateRooms, saveRooms };
