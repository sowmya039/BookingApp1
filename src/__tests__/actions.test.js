import configureMockStore from 'redux-mock-store';
import { getSavedRooms, updateRooms, saveRooms } from '../actions';
const mockStore = configureMockStore();
const store = mockStore();

describe('Action creators validity tests', () => {
    it('creates a GET_SAVED_ROOMS action', () => {
        store.dispatch(getSavedRooms());
        expect(store.getActions()).toMatchSnapshot();
    });
    it('creates a UPDATE_ROOMS action', () => {
        store.dispatch(updateRooms({}, 2));
        expect(store.getActions()).toMatchSnapshot();
    });
    it('creates a SAVE_ROOMS action', () => {
        store.dispatch(saveRooms());
        expect(store.getActions()).toMatchSnapshot();
    });
});
