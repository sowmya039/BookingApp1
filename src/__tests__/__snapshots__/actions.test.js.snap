// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Action creators validity tests creates a GET_SAVED_ROOMS action 1`] = `
Array [
  Object {
    "type": "GET_SAVED_ROOMS",
  },
]
`;

exports[`Action creators validity tests creates a SAVE_ROOMS action 1`] = `
Array [
  Object {
    "type": "GET_SAVED_ROOMS",
  },
  Object {
    "payload": Object {
      "ev": Object {},
      "index": 2,
    },
    "type": "UPDATE_ROOMS",
  },
  Object {
    "type": "SAVE_ROOMS",
  },
]
`;

exports[`Action creators validity tests creates a UPDATE_ROOMS action 1`] = `
Array [
  Object {
    "type": "GET_SAVED_ROOMS",
  },
  Object {
    "payload": Object {
      "ev": Object {},
      "index": 2,
    },
    "type": "UPDATE_ROOMS",
  },
]
`;
